scripts/cluster/join.py

```diff
@@ -302,6 +408,66 @@ def update_kubeproxy(token, ca, master_ip, api_port, hostname_override):
     service("restart", "proxy")
 
 
+def update_cert_auth_kubeproxy(token, ca, master_ip, master_port, hostname_override):
+    """
+    Configure the kube-proxy
+
+    :param token: the token to be in the kubeconfig
+    :param ca: the ca
+    :param master_ip: the master node IP
+    :param master_port: the master node port where the cluster agent listens
+    :param hostname_override: the hostname override in case the hostname is not resolvable
+    """
+    proxy_token = "{}-proxy".format(token)
+    traefik_port = get_traefik_port()
+    cert = get_client_cert(master_ip, master_port, proxy_token, "kubeproxy")
```

Comment:
This is needed for RBAC to work.

```suggestion
    cert = get_client_cert(master_ip, master_port, proxy_token, "system:kube-proxy")
```

Reply:
LGTM, but up for doing something for the filename, just in case that `:` comes back to haunt us in a few years time.

Up to you to update as you see fit.